activerecord/lib/active_record/attribute.rb

```diff
@@ -37,6 +37,9 @@ def value
       # `defined?` is cheaper than `||=` when we get back falsy values
       @value = type_cast(value_before_type_cast) unless defined?(@value)
       @value
+    rescue ActiveRecord::SerializationTypeMismatch => error
+      error.attribute = name
```

Comment:
Mutating the error is a bit of a smell to me. And, like you said, having the top level attribute know about this error seems off too.

We need to find some other way to extract this. Maybe there's some way of making the YAMLColumn aware of the attribute (or just it's column name) when it's instantiated?